### PR TITLE
Overhaul sql statement serialization - phase 1

### PR DESCRIPTION
**Pull request by Maxxen**
This is a WIP PR to get feedback and get started fixing CI. A better description + tests are coming.
Implements a SQL to JSON serialization function as part of the json extension capable of serializing SELECT statments.

### CODE REVIEW

**Comment by Maxxen** [comment]
Note to self: Serializing values with logical type `Enum` does not work since we can't serialize duckdb vectors yet.

File: src/include/duckdb/common/enums/aggregate_handling.hpp [hunk 1]
```diff
@@ -19,4 +19,6 @@ enum class AggregateHandling : uint8_t {
 	FORCE_AGGREGATES       // force aggregates: any non-aggregate select list entry will become a GROUP
 };
 
+const char *ToString(AggregateHandling value);
```
Comment (Mytherin) [inline src/include/duckdb/common/enums/aggregate_handling.hpp:22]: Could we call this something less generic - e.g. `EnumToString`?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! This looks great. Could we add a verification test similar to the `DeserializedStatementVerifier` that uses this serializer to serialize the statement and the original `Deserialize` to deserialize it? That should make sure all statements are serialized correctly.

**Comment by Mytherin** [comment]
Closing this for now as it is taking away CI resources from the release

**Comment by Maxxen** [comment]
My bad, I thought release was all done and CI was up for grabs again ;)

**Comment by Mytherin** [comment]
No worries

**Comment by Mytherin** [comment]
Even after release though it would be better not to trigger CI in the main repo unnecessarily for WIP stuff - if everyone were to do that it would always be super clogged :)

**Pull request closed without merging**